Source/CollapsedLayout.swift

```diff
@@ -23,9 +23,28 @@ public class CollapsedLayout: UICollectionViewFlowLayout {
         let insets = collectionView.contentInset
         return collectionView.bounds.width - (insets.left + insets.right)
     }
+
+    public var contentSizeAdjustmentBehavior: ContentSizeAdjustmentBehavior = .normal {
+        didSet {
+            self.invalidateLayout()
+        }
+    }
     
     override public var collectionViewContentSize: CGSize {
-        return CGSize(width: contentWidth, height: max(contentHeight, self.collectionView?.bounds.size.height ?? 0))
+        switch self.contentSizeAdjustmentBehavior {
+        case .normal:
+            return CGSize(width: self.contentWidth, height: self.contentHeight)
+        case .fitHeight(let adjustInsets):
+            var adjustedContentHeight = self.collectionView?.bounds.height ?? 0
```

Comment:
I think it would be more readable to remove optionality right at the beginning of the calculations. `contentWidth` actually does that, so we can just return `CGSize(width: self.contentWidth, height: 0)` if there is no `collectionView` (which I really doubt can happen 🤔)